core/src/main/resources/bosatsu/predef.bosatsu

```diff
@@ -67,6 +69,16 @@ def concat(front: List[a], back: List[a]) -> List[a]:
     []: front
     _: reverse_concat(reverse(front), back)
 
+def uncurry2(f: t1 -> t2 -> r) -> (t1, t2) -> r:
+  \tuple ->
+    (x1, x2) = tuple
+    f(x1, x2)
+
+def uncurry3(f: t1 -> t2 -> t3 -> r) -> (t1, t2, t3) -> r:
+  \tuple ->
+    (x1, x2, x3) = tuple
```

Comment:
I want to support patterns in lambdas so this can become `\(x1, x2, x3) -> f(x1, x2, x3)`